napalm_panos/panos.py

```diff
@@ -509,28 +524,26 @@ def get_interfaces(self):
                 interface_info_xml = xmltodict.parse(self.device.xml_root())
                 interface_info_json = json.dumps(interface_info_xml['response']['result']['hw'])
                 interface_info = json.loads(interface_info_json)
-            except AttributeError:
-                interface_info = {}
+            except KeyError:
+                # loopback sub-ifs don't return a 'hw' key
+                interface_dict[intf] = LOOPBACK_SUBIF_DEFAULTS
+                continue
 
-            name = interface_info.get('name')
-            state = interface_info.get('state')
+            state = interface_info.get('state') == 'up'
+            state_enabled = interface_info.get('state_c') != 'down'  # up -> 'up' or 'auto'
```

Comment:
seems dangerous, what about disabled or err-disabled?? again, I would rather be explicit and fail early and often. False positives are worse then failures IMHO

Reply:
`'state_c'` is the configured state, it can only be `('up', 'down', 'auto')`, for all practical purposes, `auto` is enabled.

Reply:
it's 3 lines of code, you mind doing me the flavor? Again, i worry about API changing and adding additional options down the road. Which has happened with the API before, the pa to html stuff would break between versions for this exact reason.

Reply:
not quite 3 😛 but what about:
```python
conf_state = interface_info.get('state_c')
if conf_state == 'down':
    state_enabled = False
elif conf_state in ('up', 'auto'):
    state_enabled = True
else:
    raise RuntimeError('Unknown configured state {} for interface {}'.format(conf_state, name))
```
?

Reply:
awesome!